provider/machinepool/hcp/machine_pool_resource.go

```diff
@@ -1186,7 +1190,9 @@ func populateState(ctx context.Context, object *cmv1.NodePool, state *HcpMachine
 		}
 		state.TuningConfigs = tuningConfigsList
 	} else {
-		state.TuningConfigs = types.ListNull(types.StringType)
+		if !common.HasValue(state.TuningConfigs) {
```

Comment:
Does this need to be wrapped in the else

Reply:
Gotcha, I think we need similar refactors throughout then

Reply:
not a blocker, just NP, I wouldnt roll with making more refactors, functionally its good as is